### PR TITLE
Only use schema load for Rails version <5

### PR DESCRIPTION
**Pull request by hstebbins-mdsol**
Hi folks,

This PR updates the workaround in `ci:setup_db` to address [this Rails 4 issue](https://github.com/rails/rails/issues/19001).

This change can cause a problem in CI: if there is a database migration that _doesn't_ change `schema.rb`, the database won't be in the correct state during CI.

This came up in https://github.com/mdsol/dalton/pull/3269, where we've written a migration to create database views that aren't reflected in the Rails schema. We're working around it by manually running the tasks in `ci:config` in the `travis.yml`, replacing `db:schema:load` with `db:migrate`. 

I'm not really sure how kender updates work, but i just wanted to get the ball rolling by submitting a PR.

### CODE REVIEW

**Comment by hstebbins-mdsol** [comment]
ping @jfeltesse-mdsol / @JordiPolo since it looks like y'all have been handling PRs to this repo -- let me know if I got the wrong people :)

**Comment by jfeltesse-mdsol** [comment]
Putting it simply, I'm against this change. `schema:load` is the preferred way to restore a DB and while I do understand your use case, this will have the side effect of having everyone switch to using `db:migrate` onwards. `db:migrate` is slow and tends to cause issues in very old apps with a long history of migrations, like the comment at the top of the schema file explains.

Among all the services we manage very few use views. I think in your case the better work around is to switch the schema from ruby to SQL, wouldn't this persist the view-creating migration's SQL?

**Comment by hstebbins-mdsol** [comment]
consider me convinced! thanks for the insight Julien

**Comment by hstebbins-mdsol** [comment]
Reopening because I was a bit too hopeful about SQL schema and I'm not sure where else we should have this conversation.

Using a SQL schema _does_ persist the views, but sadly it doesn't play very nice with the rest of Rails. You can read more about it [here](https://guides.rubyonrails.org/active_record_migrations.html#types-of-schema-dumps).

tl;dr: two problems: `db:migrate` doesn't regenerate the SQL schema and `db:schema:load` doesn't load it. The analogous commands are `db:structure:dump` and `db:structure:load`. 

So, I'm not really sure what the best path forward is. We'll continue with the `ci:setup_db` workaround for now. Any ideas on improving this going forward?

FYI @mdsol/team04

**Comment by jfeltesse-mdsol** [comment]
Does `db:structure:load` work in your case? If so I think the condition should be changed to like
- schema.rb? => `db:schema:load`
- schema.sql? => `db:structure:load`
- else => `db:migrate`

I think the comment about the rails 4 issue can be removed now.

If that way it works for you too and @jcarres-mdsol agrees then you also need to update the version and the changelog. If nothing helps then I guess you should not use kender, the tool is just a way to reduce copy/pasting of commands in travis, it is not meant to cover all the use cases.

**Comment by PoloPro** [comment]
Rather than update the behavior of the default `ci:config` task, it may be preferable to add an optional argument. Passing the argument would force `db:migrate` for those applications which require the workaround.

I would prefer to update kender rather than use it piecemeal, to prevent code drift between core repositories.

**Comment by ctsstc** [comment]
@jfeltesse-mdsol Thanks! The `db:schema:load` is what I needed to get my db back to sanity. 

Beforehand I did a `db:drop db:create db:migrate`, it removed all the limits from the schema and made changes a nightmare to find along with re-ordering columns.

I'm guessing that it's all dependent on Postgres version you're running, but if you have an established db state from an older version it will honor what's already there, otherwise from a fresh db it will change the schema to match the new implicit defaults that are explicitly stated.

**Comment by jcarres-mdsol** [comment]
@hstebbins-mdsol 
Is it ok to close this PR?

**Comment by DMcKinnon-mdsol** [comment]
Alas, Harry has left the company :(.

I'll leave the fate of this PR up to you all.

**Comment by jfeltesse-mdsol** [comment]
Closing because no one considers this to be the proper solution, if anything it should be something along the lines of @PoloPro's suggestion.
